src/components/Header/UpgradePlanForm/UpgradePlanForm.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import axios from 'axios';
+
+class UpgradePlanForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      userPlanModel : null,
+      formIsValid: false
+    }
+
+    this.submitForm = this.submitForm.bind(this);
+  }
+
+  componentWillMount() {
+    axios
+      .get(process.env.REACT_APP_API_URL + '/SendUpgradePlanContactEmail/GetUpgradePlanData?idUserType=' + this.props.idUserType , {
+        withCredentials: true
+      })
+      .then((response) => {
+        this.setState({ userPlanModel: response.data.data });
```

Comment:
It is not clear for me what is the meaning and structure of _userPlanModel_.

What do you think about having an explicit map between _response.data.data_ and _userPlanModel_ it will make it clear.

Reply:
because as we said, we did not spend a lot of time on this. I did it as it is done today in Doppler

Reply:
😕 We want to spend little time on it, but make it as well as possible. If we have a mapper between that, it will be easy to maintain and it will be less error-prone.